src/storage/v3/delta.hpp

```diff
@@ -27,6 +28,11 @@ struct Edge;
 struct Delta;
 struct CommitInfo;
 
+inline uint64_t GetNextDeltaUUID() noexcept {
+  static uint64_t uuid{0};
+  return ++uuid;
+}
```

Comment:
I agree I wanted to make this a shard concern, but it is difficult to propagate the id generation since it would be needed in Vertex and Edge accessors.

Reply:
I saw you made this into a `SpinLock`-ed `Synchronized`. It is a good question whether this or an atomic variable is more suitable here.

It makes testing definitely harder (no way to inject values we want to check), but feel free to keep it as is if you think it is too much hassle.